File: b_rank_skillcheck_archive/kimariji_03.cs
// 実行時間 10ms
using static System.Console;

internal class Program
{
    static void Main()
    {
        ReadLine();
        int o = 0;
        string s1 = ReadLine();
        string s2 = ReadLine();

        for( int i = 0; i < System.Math.Min( s1.Length, s2.Length ); i++ ) {
            if( s1[i] == s2[i] ) {
                o++;
            } else {
                break;
            }
        }
        WriteLine( o );
        return;
    }
}

File: b_rank_skillcheck_archive/kimariji_03.py
# 実行時間 10ms
input()
o = 0
s1 = input()
s2 = input()

for i in range(min(len(s1), len(s2))):
    if s1[i] == s2[i]:
        o += 1
    else:
        break
print(o)
